HyperVExtension/src/HyperVExtension/CommunicationWithGuest/Responses/ResponseBase.cs

```diff
@@ -100,4 +100,16 @@ protected uint GetRequiredUintValue(string valueName)
             throw new ArgumentException($"{valueName} cannot be empty.", ex);
         }
     }
+
+    protected bool GetRequiredBoolValue(string valueName)
+    {
+        try
+        {
+            return (bool?)JsonData[valueName] ?? throw new ArgumentException($"{valueName} cannot be empty.");

```

Comment:
Yeah, I think it would. When I initially wrote it for strings, null was a possible value and I added an explicit check. Then I decided to keep the same shape of this method for value types since (re-)throwing an exception with value name gives more diagnostic info (and compiler complains about "possible null value").